### PR TITLE
[INT-884] Change handling of refresh token in case of Kloudless API

### PR DESCRIPTION
**Pull request by agataHappy**
### **Story:**

https://dronedeploy.atlassian.net/browse/INT-884

### **Work done**

Checking on refresh request, in refreshHandler, if refersh_token is empty and access_token is present and its expiration date is 1970-01-01T00:00:00.000Z. It should only work for Kloudless.

### CODE REVIEW

**Comment by agataHappy** [comment]
Should I also change version?

File: handlers/authorizationCode.js [hunk 1]
```diff
@@ -89,6 +92,12 @@ const isEmptyToken = (data) => {
   return isUndefined || isEmpty;
 };
 
+const isAccessTokenValidForever = (data) => {
+  const isRefreshTokenEmpty = data.refreshToken === "";
+  const isAccessTokenGoodForever = data.accessToken !== "" && data.access_expires_at === '1970-01-01T00:00:00.000Z';
```
Comment (lucasrabiec) [inline handlers/authorizationCode.js]: Same here but `!!data.accessToken` and I think you can just name this variable as `isForeverAccessToken`

**Comment by lucasrabiec** [comment]
> Should I also change version?

@agataHappy if you added new feature with backward compatibility you should increment the middle digit and reset last digit to zero. If you fix the bug you should only increment third digit.
You handle new case with backward compatibility for Kloudless, so I think you can change version according to the above clue.

File: handlers/authorizationCode.js [hunk 2]
```diff
@@ -89,6 +92,12 @@ const isEmptyToken = (data) => {
   return isUndefined || isEmpty;
 };
 
+const isAccessTokenValidForever = (data) => {
+  const isRefreshTokenEmpty = data.refreshToken === "";
```
[the remark this answers is not in the document]
Comment (agataHappy) [reply handlers/authorizationCode.js]: Done

File: handlers/authorizationCode.js (hunk 1 above)
Comment (agataHappy) [reply handlers/authorizationCode.js]: Done

File: handlers/authorizationCode.js [hunk 3]
```diff
@@ -89,6 +92,12 @@ const isEmptyToken = (data) => {
   return isUndefined || isEmpty;
 };
 
+const isAccessTokenValidForever = (data) => {
```
Comment (mikolaj-leszczynski) [inline handlers/authorizationCode.js]: I would also change it to: 
```
const isAccessTokenValidForever = ({ refreshToken, accessToken, access_expires_at }) => {
  return !refreshToken && !!accessToken && access_expires_at === '1970-01-01T00:00:00.000Z';
};
```

Comment (agataHappy) [reply handlers/authorizationCode.js]: Done

**Pull request merged**